### PR TITLE
MULE-10866 Make templating mechanism work with ExtensionModel for its generation

### PR DESCRIPTION
**Pull request by fernandezlautaro**
Same PR as [before](https://github.com/mulesoft/mule/pull/3591), fixing the broken build due to missing dependencies (and a some tests I didn't run)

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/ModuleDelegatingEntityResolver.java [hunk 1]
```diff
@@ -35,14 +37,19 @@
...
   public ModuleDelegatingEntityResolver(Optional<ExtensionManager> extensionManager) {
     this.entityResolver = new DelegatingEntityResolver(Thread.currentThread().getContextClassLoader());
     this.extensionManager = extensionManager;
     ServiceRegistry spiServiceRegistry = new SpiServiceRegistry();
-    schemaResourceFactory =
-        spiServiceRegistry.lookupProvider(SchemaResourceFactory.class, getClass().getClassLoader());
+    try {
+      // TODO(fernandezlautaro): MULE-11024 until the implementation is moved up to extensions-api, we need to work with Optional to avoid breaking the mule testing framework (cannot add the dependency, as it will imply a circular dependency)
+      schemaResourceFactory =
+          of(spiServiceRegistry.lookupProvider(SchemaResourceFactory.class, getClass().getClassLoader()));
```
Comment (pablokraan) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/ModuleDelegatingEntityResolver.java]: LookupProvider will throw IllegalStateException when there is zero or multiple providers. Setting empty is not the proper way to manage this situation. I think that you should use getProviders instead and manage the different results

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/ModuleDelegatingEntityResolver.java [hunk 2]
```diff
@@ -56,7 +63,8 @@ public InputSource resolveEntity(String publicId, String systemId) throws SAXExc
 
   private InputSource generateModuleXsd(String publicId, String systemId) {
     InputSource inputSource = null;
-    if (extensionManager.isPresent()) {
+    if (extensionManager.isPresent()
```
Comment (pablokraan) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/ModuleDelegatingEntityResolver.java]: Is not an error if the extensionManager is absent and the schemaResourceFactory is present? (and viceversa)

Comment (fernandezlautaro) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/ModuleDelegatingEntityResolver.java]: ExtensionManager could be absent, but the schemaResourceFactory must not.
I'll add the same TODO to the `if` condition so it gets removed once MULE-11024 is done

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/ModuleDelegatingEntityResolver.java (hunk 1 above)
Comment (fernandezlautaro) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/ModuleDelegatingEntityResolver.java]: done

**Pull request merged**